galleries/examples/specialty_plots/radar_chart.py

```diff
@@ -99,15 +99,17 @@ def _gen_axes_spines(self):
                 return super()._gen_axes_spines()
             elif frame == 'polygon':
                 # spine_type must be 'left'/'right'/'top'/'bottom'/'circle'.
-                spine = Spine(axes=self,
-                              spine_type='circle',
-                              path=Path.unit_regular_polygon(num_vars))
+                polar_spine = Spine(axes=self,
+                                    spine_type='circle',
+                                    path=Path.unit_regular_polygon(num_vars))
                 # unit_regular_polygon gives a polygon of radius 1 centered at
                 # (0, 0) but we want a polygon of radius 0.5 centered at (0.5,
                 # 0.5) in axes coordinates.
-                spine.set_transform(Affine2D().scale(.5).translate(.5, .5)
-                                    + self.transAxes)
-                return {'polar': spine}
+                polar_spine.set_transform(Affine2D().scale(.5).translate(.5, .5)
+                                          + self.transAxes)
+                # Must have a dummy spine for 'inner'
+                inner_spine = Spine.linear_spine(self, 'bottom')
+                return {'polar': polar_spine, 'inner': inner_spine}
```

Comment:
Is requiring this extra 'inner' something other axes are going to need?  eg, maybe check this works w/ cartopy?

Reply:
Yeah, I'm not happy with this consequence.

I think a better solution is to split `Spline.clear` into two, so that one can avoid clearing the registered axis. In this way there will only be one clear per axis and this can be avoided.

It still opens the question as to whether the spines should have their Axis registered (as that now seems possible without any test failures).

Reply:
FWIW I just ran the Cartopy tests against this locally and they seem fine to me.
```
== 1 failed, 790 passed, 24 skipped, 2 xfailed, 2 xpassed, 8585 warnings in 84.52s (0:01:24) ==

```
(the one failure is related to #25162 I think)

Reply:
Thanks!

That probably means that we can still register the Axis on the Spines, but it would still be better to only clear the Axis once.